tests/test_plots.py

```diff
@@ -0,0 +1,193 @@
+import warnings
+from functools import partial, wraps
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import mudata as md
+import plotnine
+import pytest
+from matplotlib.testing.decorators import image_comparison as mpl_image_comparison
+from packaging.version import Version
+
+import prismo as pr
+
+image_comparison = partial(
+    mpl_image_comparison, extensions=["png"], tol=0.5
+)  # tolerance for differences in text rendering
+plotnine.options.base_family = "DejaVu Sans"  # bundled with Matplotlib
+
+
+@pytest.fixture(scope="module")
+def cll_data():
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=FutureWarning)
+        return md.read_h5mu(Path(__file__).parent / "plots" / "cll.h5mu")
+
+
+@pytest.fixture(scope="module")
+def cll_model():
+    return pr.PRISMO.load(Path(__file__).parent / "plots" / "cll_model.h5")
+
+
+@pytest.fixture(scope="module")
+def mousebrain_model():
+    return pr.PRISMO.load(Path(__file__).parent / "plots" / "mousebrain_model.h5")
+
+
+def plotnine_comparison(*decorator_args, **decorator_kwargs):
+    def wrapper(func):
+        @wraps(func)
+        def decorated(*args, **kwargs):
+            plots = func(*args, **kwargs)
+            if not isinstance(plots, list) and not isinstance(plots, tuple):
+                plots = [plots]
+            for plot in plots:
+                plt.figure(plot.draw(show=False))
+
+        return pytest.mark.xfail(
+            condition=Version(plotnine.__version__).is_prerelease, reason="plotnine pre-release", strict=False
+        )(image_comparison(*decorator_args, **decorator_kwargs)(decorated))
+
+    return wrapper
+
+
+@plotnine_comparison(baseline_images=["overview"])
+def test_overview(cll_data):
+    return pr.pl.overview(cll_data)
+
+
+@plotnine_comparison(baseline_images=["training_curve"])
+def test_training_curve(cll_model):
+    return pr.pl.training_curve(cll_model)
+
+
+@plotnine_comparison(baseline_images=["factor_correlation"])
+def test_factor_correlation(cll_model):
+    return pr.pl.factor_correlation(cll_model)
+
+
+@plotnine_comparison(baseline_images=["variance_explained"])
+def test_variance_explained(cll_model):
+    return pr.pl.variance_explained(cll_model)
+
+
+@plotnine_comparison(baseline_images=["all_weights", "all_weights_Mutations", "all_weights_Mutations_mRNA"])
+def test_all_weights(cll_model):
```

Comment:
Not sure if it's any better but you could also use `pytest.mark.parametrize` and loop that way over the parameters

Reply:
I considered that, the problem is that you need to explicitly pass the `baseline_images` file names to the Matplotlib `image_comparison` decorator, and AFAICT there is no way to pass test IDs from `parametrize` to a second decorator.